tools/java/org/unicode/cldr/util/DtdData.java

```diff
@@ -1434,12 +1438,23 @@ public int compare(String element, String attribute, String value1, String value
     };
 
     public boolean hasValue(String elementName) {
-        return nameToElement.get(elementName).type == ElementType.PCDATA;
```

Comment:
since it is a boolean, we can change instead to be

temp = nameToElement.get(elementName)
return temp != null && temp.type == ElementType.PCDATA;

(If you want other code in there for debugging, should be under a flag.)

Reply:
Didn't see response to this.

Reply:
The only change I had made was to wrap in "try" and "catch" to work around a bug, whereas your comment has more to do with the original code, which seems tangential to this ticket...

Reply:
Can't repro anymore, so I'll restore the original code. I suspect this was the same as the mysterious transient "grammaticalState" bug.